compiler/src/iree/compiler/Dialect/HAL/Transforms/test/fixup_legacy_sync.mlir

```diff
@@ -43,6 +43,28 @@ func.func @legacy_mode_not_required(%arg0: !hal.device) {
 
 // -----
 
+// Tests that any device requiring legacy_sync will trigger the pass.
+
+module attributes {hal.device.targets = [
```

Comment:
And these could be slapped at the very start of pipeline?

Reply:
yep! AssignTargetDevices does this today but a user higher up could run the pass (or do their own TargetBackend queries) whenever they want.